doc/guide/migration/mqtt.txt

```diff
@@ -0,0 +1,113 @@
+/**
+@page migration_mqtt MQTT
+@brief How to migrate an MQTT application from v3 to v4.
+
+The MQTT library has been redesigned in v4 as a thread-aware library supporting asynchronous operations. New features such as persistent session support and retry of QoS 1 publish messages have also been added.
+
+The features of the MQTT libraries in v3 and v4 are defined by the common MQTT 3.1.1 spec; therefore, the two versions are similar.
+
+@section migration_mqtt_removed Removed features
+@brief The following features which were present in v3 were removed in v4.
+- Auto-reconnect <br>
+  Version 3 has an auto-reconnect feature triggered through a call to `aws_iot_mqtt_yield`. In version 4, `aws_iot_mqtt_yield` has been removed and replaced with background tasks. <br>
+  <b>Workaround:</b> When a version 4 API call returns @ref IOT_MQTT_NETWORK_ERROR or the version 4 MQTT disconnect callback is invoked, call @ref mqtt_function_connect through the application.
```

Comment:
Please add instructions that any reconnections should be done [using exponential backoff as was done in the previous version](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/src/aws_iot_mqtt_client_yield.c#L102) since this responsibility now falls upon the API user.  More information about why this is important [here.](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter).

Reply:
added